test/test_coap_listen.py

```diff
@@ -164,26 +117,29 @@ def test_plugin_reconfigure_else(mocker, unused_port):
     mocker.patch.dict(config, {'port': port})
     config['port']['value'] = config['port']['default']
     config['uri']['value'] = config['uri']['default']
-    log_exception = mocker.patch.object(coap._LOGGER, "exception")
+    new_config = copy.deepcopy(_NEW_CONFIG)
+    new_config['port']['value'] = new_config['port']['default']
+    new_config['uri']['value'] = new_config['uri']['default']
+    pstop = mocker.patch.object(coap, 'plugin_shutdown', return_value=True)
     log_info = mocker.patch.object(coap._LOGGER, "info")
 
     # WHEN
-    loop.call_soon(coap.plugin_start(config))
-    await asyncio.sleep(.5)  # required to allow ensure_future task to complete
-    coap._plugin_stop(config)
+    new_handle = coap.plugin_reconfigure(config, new_config)
+    await asyncio.sleep(.3)
 
     # THEN
-    assert 2 == log_info.call_count
-    calls = [call('CoAP listener started on port {} with uri sensor-values'.format(config['port']['value'])),
-             call('Stopping South CoAP plugin...')]
-    log_info.assert_has_calls(calls, any_order=True)
-    assert 0 == log_exception.call_count
+    assert new_config == new_handle
+    assert 3 == log_info.call_count
```

Comment:
We should assert shutdown init and start called in sequence

Reply:
done

Reply:
plugin_shutdown() patching has been removed.